dipy/tracking/local/localtracking.py

```diff
@@ -93,12 +100,22 @@ def _generate_streamlines(self):
             directions = dg.initial_direction(s)
             directions = directions[:max_cross]
             for first_step in directions:
-                stepsF = local_tracker(dg, tc, s, first_step, vs, F, ss, fixed)
-                if stepsF < 0:
+                stepsF, tissue_class = local_tracker(dg, tc, s, first_step,
+                                                     vs, F, ss, fixed)
+                # enum TissueClass (tissue_classifier.pyd) is not accessible
+                # from here. To be changed when minimal cyhton version > 0.21.
+                # cython 0.21 - cpdef enum to export values into Python-level namespace
+                # https://github.com/cython/cython/commit/50133b5a91eea348eddaaad22a606a7fa1c7c457
+                if not (self.return_all or
+                        tissue_class == 2 or  # ENDPOINT
+                        tissue_class == -1):  # OUTSIDEIMAGE
```

Comment:
To me, we either want all generated streamlines, or only those ending with a 'ENDPOINT' or 'OUTSIDEIMAGE' state. You are right, there is a difference: 'INVALIDPOINT' produced a wrong streamline and 'TRACKPOINT' produced an incomplete streamline. 
We could change ```return_all=True``` by ``` return_valid=True,  return_invalid=True, return_incomplete=True```, so one could simply choose which streamlines he wants. :)

What do you think?